block-storage/src/main/scala/coop/rchain/blockstorage/util/io/package.scala

```diff
@@ -0,0 +1,76 @@
+package coop.rchain.blockstorage.util
+
+import java.io.IOException
+import java.nio.file._
+import java.util.stream.Collectors
+
+import cats.data.EitherT
+import cats.implicits._
+import cats.effect.Sync
+import coop.rchain.blockstorage.util.io.IOError.{IOErr, IOErrT}
+
+import scala.collection.JavaConverters._
+
+package object io {
+  private[io] def handleIo[F[_]: Sync, A](
```

Comment:
I don't think this "handleIo" is a good name.

when I read code like ```handleIo(Files.move(...), {case e: FileAlreadyExistsException => ...})``` I get the feeling that the exception is being handled.

In fact you are only wrapping&delaying the action so this is why I find the name misleading.

I would rather have something like:
```scala
object IOAction {
  def apply[A](a : => A) = {...}
}

def moveFile(...)  : F[IOAction[Path]] = IOAction(Files.move(from, to, options: _*))
```